crates/services/txpool_v2/src/selection_algorithms/mod.rs

```diff
@@ -0,0 +1,48 @@
+use std::fmt::Debug;
+
+use fuel_core_types::services::txpool::PoolTransaction;
+
+use crate::{
+    error::Error,
+    storage::StorageData,
+};
+
+pub mod ratio_tip_gas;
+
+/// Constraints that the selection algorithm has to respect.
+pub struct Constraints {
+    pub max_gas: u64,
+}
+
+pub trait SelectionAlgorithmStorage {
+    type StorageIndex: Copy + Debug;
+
+    fn get(&self, index: &Self::StorageIndex) -> Result<&StorageData, Error>;
+    fn get_dependents(
+        &self,
+        index: &Self::StorageIndex,
+    ) -> Result<Vec<Self::StorageIndex>, Error>;
+}
+
+/// The selection algorithm is responsible for selecting the best transactions to include in a block.
+pub trait SelectionAlgorithm<S: SelectionAlgorithmStorage> {
```

Comment:
Can we take the same approach as the `CollisionManager` here?

Reply:
Closing this in favor of the big thread above